### ISSUE TITLE
Error when writing HISTORY output of ASNOW from ENSAVG Gridcomp

### ISSUE THREAD

**Issue by gmao-rreichle** [GEOS-ESM/GEOSldas_GridComp#48]
@saraqzhang reported an error when HISTORY output of ASNOW is requested from the [ENSAVG Gridcomp](https://github.com/GEOS-ESM/GEOSldas_GridComp/blob/develop/GEOSens_GridComp/GEOS_EnsGridComp.F90).  This **only** works when the other fractional area coverage variables (FRSAT, FRUST, and FRWLT) are also (simultaneously) written out by HISTORY. 

If ASNOW output is requested **without** simultaneously writing FRSAT, FRUST, and FRWLT, I think the following calculation fails:
https://github.com/GEOS-ESM/GEOSldas_GridComp/blob/9d6d6997de3eacda74e213d11f10f0eb00a5f149/GEOSens_GridComp/GEOS_EnsGridComp.F90#L3686

Note that ASNOW is treated differently from the other variables:
```
    call MAPL_GetPointer(import, FRSAT,  'FRSAT' ,rc=status)
    VERIFY_(status)
    call MAPL_GetPointer(import, FRUST,  'FRUST' ,rc=status)
    VERIFY_(status)
    call MAPL_GetPointer(import, FRWLT,  'FRWLT' ,rc=status)
    VERIFY_(status)
    ! for offline model , there is no 'ASNOW', recompute 
    ! call MAPL_GetPointer(import, ASNOW,  'ASNOW' ,rc=status)
    ! VERIFY_(status)

```
https://github.com/GEOS-ESM/GEOSldas_GridComp/blob/9d6d6997de3eacda74e213d11f10f0eb00a5f149/GEOSens_GridComp/GEOS_EnsGridComp.F90#L2636C5-L2638C22

It's not clear to me exactly why ASNOW is treated differently.  The comment: `for offline model, there is no ASNOW` doesn't quite make sense to me.  ASNOW **is** an export in Catch_GridComp.

One solution may be to somehow "activate" FRSAT, FRUST, and FRWLT whenever ASNOW output is requested. 

Or perhaps we can treat ASNOW like the other variables? 

**If** the FR* variables are written simultaneously, we **should** compute ASNOW as follows to make sure that the areas sum up to 1:
https://github.com/GEOS-ESM/GEOSldas_GridComp/blob/9d6d6997de3eacda74e213d11f10f0eb00a5f149/GEOSens_GridComp/GEOS_EnsGridComp.F90#L3686

**Comment by mathomp4** [comment]
@gmao-rreichle Usually how this is handled in GEOS is by always allocating the dependent export arrays like you do for, say, `TA_enavg`:

```fortran
    call MAPL_GetPointer(export, TA_enavg, 'TA', alloc=.true., rc=status)
    VERIFY_(status)
```

So, you could mark the `FRSAT_enavg`, `FRUST_enavg`, and `FRWLT_enavg` exports as `alloc=.true.` and then the `ASNOW_enavg` calculation:

```fortran
        if(associated(FRSAT_enavg)) FRSAT_enavg = FRSAT_enavg/NUM_ENSEMBLE
        if(associated(FRUST_enavg)) FRUST_enavg = FRUST_enavg/NUM_ENSEMBLE
        if(associated(FRWLT_enavg)) FRWLT_enavg = FRWLT_enavg/NUM_ENSEMBLE
        if(associated(ASNOW_enavg)) ASNOW_enavg = max(min(1.0-(FRSAT_enavg+FRUST_enavg+FRWLT_enavg),1.0),0.0)
```
would always work because the first three would always be allocated (associated).

**Comment by gmao-rreichle** [comment]
Many thanks, @mathomp4!  I created a [pull request](https://github.com/GEOS-ESM/GEOSldas_GridComp/pull/49) based on your suggestion.

**Comment by gmao-rreichle** [comment]
addressed with #49
